src/mantine-core/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -313,6 +313,11 @@ export const MultiSelect = forwardRef<HTMLInputElement, MultiSelectProps>(
       );
     }, [searchValue]);
 
+    useDidUpdate(() => {
+      //using greater than equal to take into account creatable type.
+      if (!disabled && _value.length >= data.length) setDropdownOpened(false);
```

Comment:
This won't work for controlled input – dropdown will open when the value is set externally:

```tsx
import { useState } from 'react';
import { MultiSelect } from '@mantine/core';

function Demo() {
  const [value, setValue] = useState(null);
  return (
    <div>
      <MultiSelect data={['hello']} value={value} onChange={setValue} />
      <button type="button" onClick={() => setValue('hello')}>set value</button>
    </div>
  )
}
```

Reply:
Could you please explain, i didn't get the issue?

Reply:
I've edited the above example. Based on `useDidUpdate` hook, it will open dropdown – this should not happen.